packages/hardhat/hardhat.config.js

```diff
@@ -8,7 +8,19 @@ require('./tasks/sale-state');
 require('./tasks/mint');
 
 module.exports = {
-  solidity: '0.8.4',
+  solidity: {
```

Comment:
I think it is not necessary, I have not seen any warning.

Reply:
gonna make the change, I need to merge with main

Reply:
without the `optimizer` the test would fail telling me `Error: Transaction reverted: trying to deploy a contract whose code is too large`

Reply:
I haven't seen that warning, but asked about it and let's keep it in case it happens.

Reply:
This is the log that I get without the optimizer (that I think we should use anyway on deploy time)

```
> @ethernauts/hardhat@0.0.1 test
> hardhat test

Compiling 18 files with 0.8.9
Warning: Contract code size exceeds 24576 bytes (a limit introduced in Spurious Dragon). This contract may not be deployable on mainnet. Consider enabling the optimizer (with a low "runs" value!), turning off revert strings, or using libraries.
  --> contracts/Ethernauts.sol:11:1:
   |
11 | contract Ethernauts is ERC721Enumerable, Ownable {
   | ^ (Relevant source part starts here and spans across multiple lines).


Compilation finished successfully


  Early mint
    1) "before all" hook: deploy contract in "Early mint"

  Test events emitted
    2) "before all" hook: deploy contract in "Test events emitted"

  General
    when deploying the contract with invalid parameters
      when deploying with too many giftable tokens
        ✓ reverts (65ms)
      when deploying with too many tokens
        ✓ reverts
    when deploying the contract with valid paratemeters
      3) "before all" hook: deploy contract for "shows the correct coupon signer"

  Gift
    4) "before all" hook: deploy contract in "Gift"

  Mint
    5) "before all" hook: deploy contract in "Mint"

  Random
    6) "before all" hook: deploy contract in "Random"

  Recovery
    7) "before all" hook: deploy contract in "Recovery"

  State Changes
    8) "before all" hook: deploy contract for "contract starts in state paused"

  Transfer
    9) "before all" hook: deploy contract in "Transfer"

  Withdraw
    10) "before all" hook: deploy contract in "Withdraw"


  2 passing (2s)
  10 failing

  1) Early mint
       "before all" hook: deploy contract in "Early mint":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  2) Test events emitted
       "before all" hook: deploy contract in "Test events emitted":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  3) General
       when deploying the contract with valid paratemeters
         "before all" hook: deploy contract for "shows the correct coupon signer":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  4) Gift
       "before all" hook: deploy contract in "Gift":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  5) Mint
       "before all" hook: deploy contract in "Mint":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  6) Random
       "before all" hook: deploy contract in "Random":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  7) Recovery
       "before all" hook: deploy contract in "Recovery":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  8) State Changes
       "before all" hook: deploy contract for "contract starts in state paused":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  9) Transfer
       "before all" hook: deploy contract in "Transfer":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at runNextTicks (internal/process/task_queues.js:62:3)
      at listOnTimeout (internal/timers.js:523:9)
      at processTimers (internal/timers.js:497:7)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

  10) Withdraw
       "before all" hook: deploy contract in "Withdraw":
     Error: Transaction reverted: trying to deploy a contract whose code is too large
      at Ethernauts.constructor (contracts/Ethernauts.sol:53)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at EthModule._estimateGasAction (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:429:9)
      at HardhatNetworkProvider.request (/Users/stermi/dev/web3/ethernauts/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:106:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
      ```